File: app/src/main/java/com/semisonfire/cloudgallery/data/local/dao/PhotoDao.java
package com.semisonfire.cloudgallery.data.local.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.semisonfire.cloudgallery.data.model.Photo;

import java.util.List;

import io.reactivex.Maybe;

@Dao
public interface PhotoDao {

    @Query("SELECT * FROM photo")
    Maybe<List<Photo>> getPhotos();

    @Query("SELECT * FROM photo WHERE upload = 0")
    Maybe<List<Photo>> getUploadingPhotos();

    @Insert (onConflict = OnConflictStrategy.REPLACE)
    long insertPhoto(Photo photo);

    @Delete
    void deletePhoto(Photo photo);
}

File: app/src/main/java/com/semisonfire/cloudgallery/data/local/dao/PhotoDao.kt
package com.semisonfire.cloudgallery.data.local.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.semisonfire.cloudgallery.data.local.entity.PhotoEntity
import io.reactivex.Single

@Dao
interface PhotoDao {

    @Query("SELECT * FROM photo")
    fun getPhotos(): Single<List<PhotoEntity>>

    @Query("SELECT * FROM photo WHERE photo.upload = 0")
    fun getUploadingPhotos(): Single<List<PhotoEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(photo: PhotoEntity)

    @Query("DELETE FROM photo WHERE photo.id=:id")
    fun deleteById(id: String)

    @Delete
    fun delete(photo: PhotoEntity)
}